src/components/flex/flex_item.tsx

```diff
@@ -61,18 +96,38 @@ export const OuiFlexItem: FunctionComponent<
 > = ({
   children,
   className,
-  grow = true,
+  grow = true, // true -> use default behaviour 'flex-grow: 1'
+  shrink = true, //  true -> use default behaviour 'flex-shrink: 1'
+  basis = true, //  true -> use default behaviour 'flex-basis: 0%'
```

Comment:
```suggestion
  basis = undefined, // undefined -> preserve default behaviour
```

Reply:
In line with your previous comment, my response there and my assumption for `basis` prop in the [comment last week](https://github.com/opensearch-project/oui/pull/1126#issuecomment-1804313627). What I suggest as rule was the following:

For `basis` prop we will have:

>- if `null/undefined/false/0`: deactivate default behaviour for `flex-basis`-> apply class `ouiFlexItem--flexBasisAuto` equivalent to `flex-basis:auto`.
- if `true`: use default component behaviour -> `flex-basis: 0%`  -> inherited from styles defined in _flex_group.scss for class `.ouiFlexItem` (previous from my work ion this issue)
- if `auto, max-content, min-content or fit-content`: apply one of those properties that will equate to `flex-basis: auto`, `flex-basis: max-content`, `flex-basis: min-content` or `flex-basis: fit-content`.

Reply:
Right, I want to make sure that if we update this, existing layouts wont break. For example, if someone already has this code:

```tsx
return (
  <EuiFlexItem grow={false}>...</EuiFlexItem>
  <EuiFlexItem grow={true}>...</EuiFlexItem>
  <EuiFlexItem grow={2}>...</EuiFlexItem>
);
```

it won't break with these changes. To me, that means that the applied styles shouldn't change with the same inputs. Only when you specify new props (i.e. specifying `shrink` or `basis`) should different styles be applied. What I'm getting at is that new style semantics should only be applied if the builder specifies `shrink` or `basis`.